sympy/matrices/dense.py

```diff
@@ -1386,33 +1387,36 @@ def randMatrix(r, c=None, min=0, max=99, seed=None, symmetric=False,
     >>> A == randMatrix(3, seed=1)
     True
     >>> randMatrix(3, symmetric=True, percent=50) # doctest:+SKIP
-    [0, 68, 43]
-    [0, 68,  0]
-    [0, 91, 34]
+    [77, 70,  0],
+    [70,  0,  0],
+    [ 0,  0, 88]
     """
     if c is None:
         c = r
     # Note that ``Random()`` is equivalent to ``Random(None)``
     prng = prng or random.Random(seed)
-    if symmetric and r != c:
-        raise ValueError(
-            'For symmetric matrices, r must equal c, but %i != %i' % (r, c))
+
     if not symmetric:
         m = Matrix._new(r, c, lambda i, j: prng.randint(min, max))
-    else:
-        m = zeros(r)
-        for i in range(r):
-            for j in range(i, r):
-                m[i, j] = prng.randint(min, max)
-        for i in range(r):
-            for j in range(i):
-                m[i, j] = m[j, i]
-    if percent == 100:
-        return m
-    else:
-        z = int(r*c*percent // 100)
+        if percent == 100:
+            return m
+        z = int(r * c * (100 - percent) // 100)
```

Comment:
As noted below, I think this should be `z = int(r * c * (100 - percent) /S(100))` and a test for percent=95 and a 10x10 matrix should be added.

Reply:
make that 99.5% with a 10x10

Reply:
What is the correct answer for 99.5? One or no zeroes? Current behaviour is no zeroes.

Reply:
Sorry -- this is a goose chase. I now believe that what you have is right.